src/tox/session.py

```diff
@@ -290,13 +290,12 @@ def verbosity(self):
         else:
             return Verbosity.DEBUG
 
-    def logpopen(self, popen, env):
+    def logpopen(self, popen, cmd_args_shell):
         """ log information about the action.popen() created process. """
-        cmd = " ".join(map(str, popen.args))
         if popen.outpath:
-            self.verbosity1("  {}$ {} >{}".format(popen.cwd, cmd, popen.outpath))
+            self.verbosity1("  {}$ {} >{}".format(popen.cwd, cmd_args_shell, popen.outpath))
         else:
-            self.verbosity1("  {}$ {} ".format(popen.cwd, cmd))
+            self.verbosity1("  {}$ {} ".format(popen.cwd, cmd_args_shell))
```

Comment:
oh nice, this is the shlex fix in the output?

Reply:
yeah should be 👍